operator/src/main/java/oracle/kubernetes/operator/helpers/CallBuilder.java

```diff
@@ -304,12 +304,63 @@
 
   private final String resourceVersion = "";
 
```

Comment:
I'm a bit troubled by adding this functionality into CallBuilder, as it seems to be something of a violation of the [Single Responsibility Principal](https://stackify.com/solid-design-principles/). Our convention has generally been to use CallBuilder simply as an interface to the Kubernetes API and put additional resource-specific behavior into a xxxHelper class. The calls to listDomain and readDomain would then go through the new DomainHelper.